databases/week3/hw.sql

```diff
@@ -0,0 +1,217 @@
+    contact_name VARCHAR(255),
+    contact_email VARCHAR(255),
+    Foreign Key (meal_id) REFERENCES Meal(id) ON DELETE SET NULL
+)
+
+CREATE Table Review(
+    id int AUTO_INCREMENT PRIMARY KEY,
+    title VARCHAR(255),
+    description VARCHAR(255),
+    meal_id int,
+    stars int,
+    created_date date,
+    Foreign Key (meal_id) REFERENCES Meal(id) ON DELETE SET NULL
+)
+
+INSERT INTO Meal (title, description, `location`, `when`, max_reservations, price, created_date) VALUES
+('Italian Pasta Night', 'Enjoy a delicious homemade Italian pasta with fresh ingredients.', 'New York, NY', '2024-03-01 19:00:00', 20, 15.99, '2024-02-15'),
+('Sushi Making Class', 'Learn how to make sushi rolls with a professional chef.', 'San Francisco, CA', '2024-03-05 18:30:00', 15, 25.00, '2024-02-18'),
+('Vegan Feast', 'A healthy and delicious vegan buffet with organic ingredients.', 'Los Angeles, CA', '2024-03-10 12:00:00', 30, 20.50, '2024-02-20'),
+('BBQ Ribs Night', 'Classic American BBQ ribs served with sides and drinks.', 'Austin, TX', '2024-03-15 20:00:00', 25, 18.75, '2024-02-22'),
+('French Pastry Workshop', 'Master the art of making croissants and éclairs.', 'Paris, France', '2024-03-20 14:00:00', 10, 30.00, '2024-02-25'),
+('Indian Curry Night', 'Enjoy authentic Indian curry with freshly baked naan.', 'London, UK', '2024-03-25 19:30:00', 20, 22.99, '2024-02-28'),
+('Mediterranean Brunch', 'A flavorful brunch with olives, hummus, and grilled meats.', 'Barcelona, Spain', '2024-03-30 11:00:00', 35, 17.49, '2024-03-01');
+
+-- Queries to write
+------------
+---Meal-----
+-----------
+
+-- Get all meals
+SELECT * FROM Meal;
+
+-- Add a new meal
+INSERT INTO Meal (title, description, `location`, `when`, max_reservations, price, created_date) VALUES
+('Japanese Ramen Night', 'Experience authentic Japanese ramen with a variety of toppings.', 'Tokyo, Japan', '2024-04-05 19:00:00', 25, 16.50, '2024-03-05');
+
+-- Get a meal with any id, fx 1
+SELECT * FROM Meal WHERE id = 1;
+
+-- Update a meal with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE Meal 
+SET title = "Pasta", description = "Enjoy a delicious homemade pasta", price = 17.50 
+    WHERE id = 1;
+
+-- Delete a meal with any id, fx 1
+DELETE FROM Meal WHERE id = 1;
+
+INSERT INTO Meal (title, description, `location`, `when`, max_reservations, price, created_date) 
+VALUES ('Italian Pasta Night', 'Enjoy a delicious homemade Italian pasta with fresh ingredients.', 'New York, NY', '2024-03-01 19:00:00', 20, 15.99, '2024-02-15'),
+
+----------------
+---Reservation----
+-----------------
+
+INSERT INTO Reservation (number_of_guests, meal_id, created_date, contact_phonenumber, contact_name, contact_email)
+VALUES 
+(4, 1, '2025-02-18', '+4522334455', 'John Doe', 'john@example.com'),
+(2, 3, '2025-02-19', '+4533445566', 'Emma Smith', 'emma@example.com'),
+(6, 4, '2025-02-20', '+4544556677', 'Michael Johnson', 'michael@example.com'); -- NULL means the meal was deleted
```

Comment:
Suggestion: Explicit is better than implicit. I mean including the NULL value on each insertion